framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3296,11 +3296,17 @@ NonlinearSystemBase::computeScaling()
 
   if (jac_scaling)
   {
-    if (!_auto_scaling_initd)
+    // if (!_auto_scaling_initd)
+    // We need to reinit this when the number of dofs changes
+    // but there is no good way to track that
+    // In theory, it is the job of libmesh system to track this,
+    // but this special matrix is not owned by libMesh system
+    // Let us reinit eveytime since it is not expensive
```

Comment:
:+1: this makes me think that I currently have no tests combining adaptivity and automatic scaling. Even more strange that no other users have hit this.